src/branch.rs

```diff
@@ -138,8 +138,8 @@ macro_rules! alt (
       match $subrule!( $i, $($args)* ) {
         $crate::IResult::Done(i,o)     => $crate::IResult::Done(i,$gen(o)),
         $crate::IResult::Incomplete(x) => $crate::IResult::Incomplete(x),
-        $crate::IResult::Error(_)      => {
-          alt!(__impl $i, $($rest)*)
+        $crate::IResult::Error(_alt_err)      => {
+          $crate::propagate_error_type(_alt_err, alt!(__impl $i, $($rest)*))
```

Comment:
I'm a bit uneasy about this. Especially, I don't know how it will interact with the whitespace parsing combinators